src/Background/Portal.vala

```diff
@@ -3,58 +3,69 @@
  * SPDX-License-Identifier: LGPL-2.1-or-later
  */
 
+[DBus (name = "org.pantheon.gala.DesktopIntegration")]
+private interface Gala.DesktopIntegration : Object {
+    public signal void running_applications_changed ();
+
+    public const string NAME = "org.pantheon.gala";
+    public const string PATH = "/org/pantheon/gala/DesktopInterface";
+
+    public struct RunningApplications {
+        string app_id;
+        HashTable<string,Variant> details;
+    }
+
+    public abstract async RunningApplications[] get_running_applications () throws DBusError, IOError;
+}
+
 [DBus (name = "org.freedesktop.impl.portal.Background")]
 public class Background.Portal : Object {
     public signal void running_applications_changed ();
 
+    private Gala.DesktopIntegration? desktop_integration;
     private DBusConnection connection;
-    private DesktopIntegration desktop_integration;
 
     public Portal (DBusConnection connection) {
         this.connection = connection;
-        NotificationRequest.init (connection);
-        try {
-            desktop_integration = connection.get_proxy_sync ("org.pantheon.gala", "/org/pantheon/gala/DesktopInterface");
-            desktop_integration.running_applications_changed.connect (() => running_applications_changed ());
-        } catch {
-            warning ("Cannot connect to compositor, background portal working with reduced functionality.");
-        }
-    }
-
-    [DBus (name = "org.pantheon.gala.DesktopIntegration")]
-    public interface DesktopIntegration : Object {
-        public struct RunningApplications {
-            string app_id;
-            HashTable<string,Variant> details;
-        }
 
-        public signal void running_applications_changed ();
-        public abstract RunningApplications[] get_running_applications () throws DBusError, IOError;
+        connection.get_proxy.begin<Gala.DesktopIntegration> (
+            Gala.DesktopIntegration.NAME,
+            Gala.DesktopIntegration.PATH,
+            NONE, null, (obj, res) => {
+                try {
+                    desktop_integration = connection.get_proxy.end (res);
+                    desktop_integration.running_applications_changed.connect (() => running_applications_changed ());
+                } catch {
+                    warning ("Cannot connect to compositor, portal working with reduced functionality.");
+                }
+            }
+        );
     }
 
+    [DBus (signature = "u")]
```

Comment:
For some reason this doesn't work here either, so it always sends a notification even with an open window. Maybe use `[CCode (type_signature = "u")]` as suggested by @tintou, I haven't tested whether that works though.